program/Project Descriptions and Plans/CORE-V-HAL/Project-Concept-HAL.md

```diff
@@ -0,0 +1,155 @@
+| 				| S-4     	|HAL Specification MUST be published and managed by an industry open source or standards body, such as OpenHW, RISC-V International, or FOSSi |
+| *Implementation Rqmtns* 	| IM-1     	|A complete open source reference implementation of the HAL for CORE-V MUST be a project deliverable |
+| 				| IM-2     	|FreeRTOS SHOULD be supported in the open source reference implementation of HAL for CORE-V |
+| 				| IM-3     	|Newlib SHOULD be supported in the open source reference implementation  of HAL for CORE-V |
+| 				| IM-3     	|The HAL reference implementation for CORE-V SHOULD be published by the body publishing the Specification (previous page) |
+  	
+
+
+### Potential future design enhancements
+- Support new peripherals or other Hardware components.
+- Create and provide reference designs on application areas such as Neural Network, DSP, Debug & Trace and Security.
+
+## Who would make use of OpenHW output
+CORE-V HAL will be used by firmware, middleware and application developers who are developing code supporting Core-V based hardware designs.
+
+It will be a key component of a Core-V MCU SDK.
+
+ 
+## Initial Estimate of Timeline
+The project will be last for 3 Months and have below output at different stage.
+
+- To agree the key technical requirements, approach, and specification for a Core-V HAL that will be re-used for all Core-V Cores
+
+Once the HAL approach is agreed
+   - To implement basic peripherals (UART, TIME, INT...) on Core-V MCU.
+   - To have FreeRTOS run  on Core-V MCU in conjunction with HAL SDK support 
+   - To implement support other peripherals and meet application requirement.
+
+## Explanation of why OpenHW should do this project
+
+- The HAL is needed to establish a baseline approach for addressing the Core-V Cores and peripherals under a consistent approach. It is a cornerstone of all Core-V SDK and software work.
+
+- There are some challenges for Core-V MCU users to develop application bases on adopted PULP SDK due to the complexity of the development configuration environment. The PULP SDK is not viewed as a basis for future development work. 
+
+- There should be an out-of-box SDK solution offered by OpenHW supporting Core-V MCU platform developer.
+
+- An easy-to-use SDK enhances the Core-V MCU development environment by reducing the barrier on software development and enabling to create reusable application code.
+
+- There are some requirement from system tools (such as compiler) to help verify their design's performance and correctness using HAL SDK.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+There are also some similar standards in market currently, the most representative HAL is below:
+   1. CMSIS for ARM cortex A and M architecture
+   2. NMSIS for RISC-V architecture
+   3. CSI for RISC-V architecture
+   4. FreeRTOS CommonIO specification https://developer.amazon.com/acs-devices
+
```

Comment:
Can we add links for the others.

Reply:
1. CMSIS: https://www.arm.com/why-arm/technologies/cmsis
2. NMSIS: https://doc.nucleisys.com/nmsis/
3. CSI: English version is under translating.

Reply:
We don't need to decide this at Project Concept stage. That is part of the next  phase, "Project Launch", when we are getting to more detail.  However, we could say something like "By Project Launch stage, we shall have decided which peripherals and core features will be developed in the first phase of the project".

Reply:
@shangyunhai For CSI, we can provide a link to the GitHub repo and the Chinese documentation and say that an English translation is in preparation.

Reply:
OK